### PR TITLE
AOP2-49 create referral

### PR DESCRIPTION
**Pull request by jasonfungsing**

### CODE REVIEW

File: frontend/apps/orams/actions/referralActions.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { SENDING_REQUEST, SET_ERROR_MESSAGE } from 'orams/constants/constants'
+import { GENERAL_ERROR } from 'orams/constants/messageConstants'
+import dmapi from 'orams/services/apiClient'
+import { setSuccessMessage } from './editPriceActions'
+
+export const sendingRequest = sending => ({ type: SENDING_REQUEST, sending })
```
Comment (qimyi) [inline frontend/apps/orams/actions/referralActions.js]: you don't need to export this fn

**Comment by waleedusman** [comment]
do you think there would be any benefit in writing any tests for this block of work?

File: frontend/apps/orams/actions/referralActions.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { setSuccessMessage } from './editPriceActions'
+
+const sendingRequest = sending => ({ type: SENDING_REQUEST, sending })
+
+const setErrorMessage = errorMessage => ({
+  type: SET_ERROR_MESSAGE,
+  errorMessage
+})
+
+export const createReferral = data => dispatch => { //eslint-disable-line
```
Comment (bsigney) [inline frontend/apps/orams/actions/referralActions.js]: what eslint rule are you disabling for this line?

File: frontend/apps/orams/components/ReferralBuilderForm/ReferralBuilderForm.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import LoadingIndicatorFullPage from 'shared/LoadingIndicatorFullPage/LoadingIndicatorFullPage'
+import Textfield from 'shared/form/Textfield'
+import formProps from 'shared/form/formPropsSelector'
+import { Form } from 'react-redux-form'
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilderForm/ReferralBuilderForm.js]: minor: put import libraries before relative/absolute imports

Comment (jasonfungsing) [reply frontend/apps/orams/components/ReferralBuilderForm/ReferralBuilderForm.js]: done

File: frontend/apps/orams/actions/referralActions.js (hunk 2 above)
Comment (jasonfungsing) [reply frontend/apps/orams/actions/referralActions.js]: this is the only function and export, we will need to add more functions later for this ticket. That's why disable eslint for this atm

File: frontend/apps/orams/components/ReferralBuilderForm/ReferralBuilderForm.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    handleCreateReferralSubmit: PropTypes.func.isRequired
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {}
+  }
+
+  retrieveRegionName() {
+    /* eslint-disable no-restricted-syntax */
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilderForm/ReferralBuilderForm.js]: did you want to turn this rule off just for this block? 
cause this rule is turned off for the whole file now.

if so you need to re-enable the rule after your code block

File: frontend/apps/orams/components/ResultsTable/ResultsTable.js [hunk 5]
```diff
@@ -20,46 +19,51 @@ class ResultsTable extends Component {
...
                 </div>
-                {category.suppliers.map((supplier, id = uniqueID()) =>
-                  <div key={id} className={styles.tableRow}>
+                {category.suppliers.map((supplier, uniqueId = uniqueID()) =>
+                  <div key={uniqueId} className={styles.tableRow}>
                     <div className="row">
                       <div className="col-xs-12 col-sm-12">
                         <span className={styles.name}>
-                          <a
+                          <a // eslint-disable-line
```
Comment (bsigney) [inline frontend/apps/orams/components/ResultsTable/ResultsTable.js]: have disabled the rule for this in .eslintrc on branch `AOP2-34-search-users`

File: frontend/apps/orams/components/ReferralBuilderForm/ReferralBuilderForm.js (hunk 4 above)
Comment (jasonfungsing) [reply frontend/apps/orams/components/ReferralBuilderForm/ReferralBuilderForm.js]: done

**Pull request merged**